permissive-json-pointer/src/lib.rs

```diff
@@ -0,0 +1,783 @@
+#![doc = include_str!("../README.md")]
+
+use std::collections::HashSet;
+
+use serde_json::*;
+
+type Document = Map<String, Value>;
+
+const SPLIT_SYMBOL: char = '.';
+
+/// Returns `true` if the `selector` match the `key`.
+/// Example:
+/// `animaux`           match `animaux`
+/// `animaux.chien`     match `animaux`
+/// `animaux.chien`     match `animaux`
+/// `animaux.chien.nom` match `animaux`
+/// `animaux.chien.nom` match `animaux.chien`
+/// -----------------------------------------
+/// `animaux`          doesn't match `animaux.chien`
+/// `animaux.`    doesn't match `animaux`
+/// `animaux.ch`          doesn't match `animaux.chien`
+/// `animau` doesn't match `animaux`
```

Comment:
```suggestion
/// Returns `true` if the `selector` match the `key`.
///
/// ```
/// Example:
/// `animaux`           match `animaux`
/// `animaux.chien`     match `animaux`
/// `animaux.chien`     match `animaux`
/// `animaux.chien.nom` match `animaux`
/// `animaux.chien.nom` match `animaux.chien`
/// -----------------------------------------
/// `animaux`          doesn't match `animaux.chien`
/// `animaux.`    doesn't match `animaux`
/// `animaux.ch`          doesn't match `animaux.chien`
/// `animau` doesn't match `animaux`
/// ```
```

Reply:
You can't do that because then it's considered a test.
And you can't write a real test with `assert_eq` either because the function isn't public.
Thus I made the block a text block instead of a code block.